app/build.gradle

```diff
@@ -40,14 +40,14 @@ android {
 dependencies {
     // There is a better way to declare and structure dependencies.
     // However, that is a different topic.
-    def daggerVersion = '2.14'
-    def butterKnifeVersion = '8.8.0'
+    def daggerVersion = '2.14.1'
+    def butterKnifeVersion = '8.8.1'
 
     annotationProcessor "com.google.dagger:dagger-compiler:$daggerVersion"
     annotationProcessor "com.google.dagger:dagger-android-processor:$daggerVersion"
     annotationProcessor "com.jakewharton:butterknife-compiler:$butterKnifeVersion"
 
-    compile "com.google.dagger:dagger:$daggerVersion"
-    compile "com.google.dagger:dagger-android:$daggerVersion"
-    compile "com.jakewharton:butterknife:$butterKnifeVersion"
+    implementation "com.google.dagger:dagger:$daggerVersion"
```

Comment:
Starting with Android Gradle plugin v3.0.0, the new [Java Library plugin configurations](https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_configurations_graph) introduced in Gradle 3.4 (`implementation` instead of `compile` are among one of them) are used. 

> For more info, see https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html#new_configurations.